### PR TITLE
Add soundpack support

### PR DESCRIPTION
**Pull request by remyroy**
This implements #14127.
- [x] Add a new option for selecting the soundpack with listing the available soundpacks.
- [x] Move our default soundpack into its own folder with the proper soundpack structure.
- [x] Modify our sound support to use the selected soundpack values instead of the hardcoded ones.
- [x] Update documentation regarding soundpack structure and directory.

### CODE REVIEW

**Comment by remyroy** [comment]
This should be complete.

**Comment by illi-kun** [comment]
Please remove [WIP] tag if it's really done.

**Comment by remyroy** [comment]
My bad. It's removed.

**Comment by illi-kun** [comment]
Nice job, BTW!

File: src/options.cpp [hunk 1]
```diff
@@ -628,6 +629,75 @@ std::string options_manager::build_tilesets_list()
     return tileset_names;
 }
 
+/** Fill SOUNDPACKS mapping with values.
+ * Scans all directores in FILENAMES["sounddir"] directory for file named FILENAMES["soundpack-conf"].
+ * All founded values added in mapping SOUNDPACKS as name, soundpack_dir.
+ * Furthermore, it builds possible values list for cOpt class.
+ * @return One string containing all found soundpacks in form "soundpack1,soundpack2,soundpack3,..."
+ */
+std::string options_manager::build_soundpacks_list()
```
Comment (BevapDin) [inline src/options.cpp]: This looks like a copy of the function for the tilesets. Is there so much difference between those two function? They seem to behave almost identical, only using different input / output variables. There should probably be one generic function that is called for both.

**Pull request merged**